File: examples/java/src/main/java/test/Quantity.java
package test;

import com.qualityminds.lazyval.LazyValue;
// tag::docu[]
@LazyValue()
// tag::docu-motivation[]
public record Quantity(int value) {

    public Quantity {
        if(value <= 0){
            throw new IllegalArgumentException("Quantity must be greater than 0");
        }
    }
}
// end::docu-motivation[]
// end::docu[]

File: examples/kotlin/src/main/kotlin/test/Quantity.kt
package test

import com.qualityminds.lazyval.LazyValue
// tag::docu[]
@LazyValue
// tag::docu-motivation[]
data class Quantity(val value: Int){
    init {
        require(value > 0) { "Quantity must be greater than 0" }
    }
}
// end::docu-motivation[]
// end::docu[]
